test/TestInternalGrains/HashBasedPlacementGrain.cs

```diff
@@ -0,0 +1,9 @@
+﻿using Orleans.Placement;
+
+namespace UnitTests.Grains
+{
+    [HashBasedPlacement(true)]
```

Comment:
since your only test is using `sortSiloList: true` (which is the default), should you kill the code that adds internal members and confusing branches for the case that you are not sorting the lists?